azure/terraform/infra/credentials.tf

```diff
@@ -6,10 +8,12 @@
 # E.g. an end user needs to be able to access Metaflow storage AND submit jobs to AKS (possibly)
 resource "azuread_application" "service_principal_application" {
   display_name = var.service_principal_name
+  owners       = [data.azuread_client_config.current.object_id]
```

Comment:
If this is not assigned there will be no owner of the service principal which requires a bit more permissions for the user that created it to be able to remove it.